src/lib/Output/Generator/Json/ArrayObject.php

```diff
@@ -4,42 +4,42 @@
  * @copyright Copyright (C) Ibexa AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
  */
+declare(strict_types=1);
 
 namespace Ibexa\Rest\Output\Generator\Json;
 
+use AllowDynamicProperties;
 use ArrayObject as NativeArrayObject;
+use Ibexa\Rest\Output\Generator\Data\DataObjectInterface;
 
 /**
  * Json array object.
  *
  * Special JSON array object implementation, which allows to access the
  * parent object it is assigned to again.
  */
-class ArrayObject extends NativeArrayObject
+#[AllowDynamicProperties]
+class ArrayObject extends NativeArrayObject implements DataObjectInterface
 {
     /**
      * Reference to the parent node.
-     *
-     * @var \Ibexa\Rest\Output\Generator\Json\JsonObject
      */
-    protected $_ref_parent;
+    protected ?DataObjectInterface $_ref_parent;
```

Comment:
As I understand, we're keeping this incorrect parameter name due to BC on the REST output, right?

Reply:
I think it's more about a potential collision with an actual `parent` property.

This issue would not exist if we had a dedicated property to stole all... well... properties (REST fields). Alas, as discussed internally, we wanted to keep the amount of changes limited.